src/App.tsx

```diff
@@ -1,24 +1,160 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { FilterBy } from './utils/enums';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoPatch } from './types/TodoPatch';
 
-const USER_ID = 0;
+const USER_ID = 10897;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [errorText, setErrorText] = useState<null | string>(null);
+
+  const resetError = () => {
+    setErrorText(null);
+  };
+
+  const setError = (errorTxt: string) => {
+    setErrorText(errorTxt);
+    setTimeout(resetError, 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch((error) => {
+        setError(error.message);
+      });
+  }, []);
+
+  const onFilterChange = useCallback((filter: FilterBy) => {
+    setFilterBy(filter);
+  }, []);
+
+  const onAddTodo = async (title: string) => {
+    if (!/\S/g.test(title)) {
+      setError('Title can\'t be empty');
+
+      return false;
+    }
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      const addedTodo = await addTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+    }
+
+    return true;
+  };
+
+  const onDeleteTodo = useCallback(async (todoId: number) => {
+    setLoadingTodoIds(prevIds => [...prevIds, todoId]);
+
+    try {
+      await deleteTodo(todoId);
+
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
```

Comment:
The `catch` block is missing an `error` parameter. 
This could be useful for logging or displaying a more specific error message. 
It should look like this: `catch (error)`.